plutus-core-spec/figures/PlutusCoreReduction.tex

```diff
@@ -144,27 +143,27 @@
 
 \begin{figure*}[t]
     \judgmentdef{\(\cksteps{\sigma}{\sigma'}\)}{Machine state $\sigma$ transitions in one step to machine state $\sigma'$}
-    
+
     \begin{align*}
+        &\cksteps{\ckforward{s}{\con{cn}}}{\ckbackward{s}{\con{cn}}}\\
         &\cksteps{\ckforward{s}{\abs{\alpha}{K}{M}}}{\ckbackward{s}{\abs{\alpha}{K}{M}}}\\
         &\cksteps{\ckforward{s}{\inst{M}{A}}}{\ckforward{s, \inInstLeftFrame{A}}{M}}\\
         &\cksteps{\ckforward{s}{\wrap{\alpha}{A}{M}}}{\ckforward{s, \inWrapRightFrame{\alpha}{A}}{M}}\\
         &\cksteps{\ckforward{s}{\unwrap{M}}}{\ckforward{s, \inUnwrapFrame{}}{M}}\\
         &\cksteps{\ckforward{s}{\lam{x}{A}{M}}}{\ckbackward{s}{\lam{x}{A}{M}}}\\
         &\cksteps{\ckforward{s}{\app{M}{N}}}{\ckforward{s, \inAppLeftFrame{N}}{M}}\\
-        &\cksteps{\ckforward{s}{\con{cn}}}{\ckbackward{s}{\con{cn}}}\\
-        &\cksteps{\ckforward{s}{\error{S}}}{\ckerror{}}\\
+        &\cksteps{\ckforward{s}{\builtin{bn}{\repetition{A}}{}}}{U} \qquad \textit{$bn$ computes on $\repetition{A}$ to $U$ according to \ref{fig:Plutus_core_builtins}}\\
```

Comment:
I'm not sure how to read this. A built-in is only applied to a spice of types, but is able to reduce somehow?

Reply:
A builtin applied to an empty spine of terms, so it can compute immediately because all of its arguments have finished computing, whereas a builtin applied to a non-empty spine of terms must begin to compute its spine. I could separate out the transition system into two separate judgments (one for term computation and one for spin computation) if people think that's clearer?

Reply:
> because all of its arguments have finished computing

But a built-in does need its arguments in order to compute? And they are just not there.

However, I see how this can work for built-ins that do not receive any arguments.

> I could separate out the transition system into two separate judgments (one for term computation and one for spin computation) if people think that's clearer?

That could be clearer, but I need to play with this and let's not block the PR on this.